src/raster/rasterband.rs

```diff
@@ -847,6 +848,87 @@ impl<'a> RasterBand<'a> {
             max: min_max[1],
         })
     }
+
+    /// Fetch default raster histogram.
+    ///
+    /// # Arguments
+    ///
+    /// * `force` - If `true`, force the computation. If `false` and no default histogram is available, the method will return None
+    pub fn default_histogram(&self, force: bool) -> Result<Option<Histogram>> {
+        let mut counts = std::ptr::null_mut();
+        let mut min = 0.0;
+        let mut max = 0.0;
+        let mut n_buckets = 0i32;
+
+        let rv = unsafe {
+            GDALGetDefaultHistogram(
+                self.c_rasterband,
+                &mut min,
+                &mut max,
+                &mut n_buckets,
+                &mut counts as *mut *mut i32,
+                libc::c_int::from(force),
+                None,
+                std::ptr::null_mut(),
+            )
+        };
+
+        match CplErrType::from(rv) {
+            CplErrType::None => Ok(Some(Histogram {
+                min,
+                max,
+                n_buckets: n_buckets as usize,
+                counts: HistCounts::Borrowed(counts),
+            })),
+            CplErrType::Warning => Ok(None),
+            _ => Err(_last_cpl_err(rv)),
+        }
+    }
+
+    /// Compute raster histogram.
+    ///
+    /// # Arguments
+    ///
+    /// * `min` - Histogram lower bound
+    /// * `max` - Histogram upper bound
+    /// * `n_buckets` - Number of buckets in the histogram
+    /// * `include_out_of_range` - if `true`, values below the histogram range will be mapped into the first bucket, and values above will be mapped into the last one. If `false`, out of range values are discarded
+    /// * `is_approx_ok` - If an approximate, or incomplete histogram is OK
+    pub fn histogram(
+        &self,
+        min: f64,
+        max: f64,
+        n_buckets: usize,
+        include_out_of_range: bool,
+        is_approx_ok: bool,
+    ) -> Result<Option<Histogram>> {
```

Comment:
Can we document what returning `Ok(None)` means?

Reply:
I really hate how these `bForce` parameters work. If I'm reading the code right, even with `bForce` you can still get a saved histogram from the PAM dataset (`.aux.xml`). I wish `!bForce` meant "return cached or compute it" and `bForce` meant "always recompute".

Reply:
@lnicola: I guess we could force it if we get None but that would diverge from the C API.

Reply:
No, let's follow the C API. I just like to complain.